mlx/treemap/treemap_directive.py

```diff
@@ -52,7 +52,21 @@ def perform_replacement(self, app):
             hash_value = sha256(str(all_data).encode()).hexdigest()  # create hash value based on graph parameters
             rel_file_path = Path('_images') / f'treemap-{hash_value}.png'
             if str(rel_file_path) not in env.images:
-                fig.write_image(str(Path(env.app.srcdir) / rel_file_path), scale=2)
+                try:
+                    fig.write_image(str(Path(env.app.srcdir) / rel_file_path), scale=2)
+                except Exception as e:
+                    # If Chrome is not installed, try to install it and retry
```

Comment:
Isn't this a bit invasive? I would assume warning and advice is more user friendly, or is this how guys at Plotly envisioned it?

Reply:
It was preinstalled in the past. I read their release notes https://github.com/plotly/Kaleido/releases/tag/v1.0.0 and their documentation and opened a bug report https://github.com/plotly/plotly.py/issues/5272.

I don't like the use of `subprocess`, but I hope it is temporary.

Kaleido and/or Chrome is only needed for PDF doc builds.

Reply:
I'll revert this workaround. The error handling by plotly is fine:

```
Traceback
=========

      File "/home/developer/venv/lib/python3.11/site-packages/sphinx/events.py", line 423, in emit
        raise ExtensionError(
    sphinx.errors.ExtensionError: Handler <function process_item_nodes at 0x7b4b4796e980> for event 'doctree-resolved' threw an exception (exception: 
    
    Kaleido requires Google Chrome to be installed.
    
    Either download and install Chrome yourself following Google's instructions for your operating system,
    or install it from your terminal by running:
    
        $ plotly_get_chrome
    
    )
 ```
 I'll allow `kaleido<1.0` and keep using that in our Docker environment.